docs/zk-passport/guide-setting-up-cross-chain-replication.mdx

```diff
@@ -0,0 +1,54 @@
+---
+description: Set up ZK Passport Registry replication to verify ZK Passport proofs on any EVM-compatible chain.
+sidebar_label: "Guide: setting up cross-chain replication"
+---
+
+import OutLink from "@site/src/components/OutLink";
+import IdealImage from '@site/src/components/IdealImage';
+
+# Guide: Setting up cross-chain replication for ZK Passport
+
+:::info
+Native state transfer to Ethereum L1 is currently under construction.
+:::
+
+This guide walks you through the process of setting up cross-chain replication for ZK Passport Registry state. By the end of this tutorial, you will have:
+
+1. Deployed the **RegistrationSMTReplicator** contract on the target chain.
+1. Set up your relayer or front-end submission flow.
+
+## Step 1: Deploy the RegistrationSMTReplicator
+
+The **RegistrationSMTReplicator** handles the replication of state from the ZK Registry onto your chain. This ensures your chain is kept up to date with changes to passport registrations.
+
+1. **Obtain the contract code** for `RegistrationSMTReplicator` (contact the provider of the ZK Registry or check their repository for the contract source).
+
+1. **Deploy** the `RegistrationSMTReplicator` to your chain:
+
+   ```solidity
+   // Example snippet (pseudo-code)
+   RegistrationSMTReplicator replicator = new RegistrationSMTReplicator(
+       registryRoot,
+       allowedUpdaterAddress
+   );
+   ```
+
+## Step 2: Set up the relayer or front-end submission flow to replicate the states
+
+To keep the state of the ZK Registry in sync with your chain, you need to set up a relayer or front-end submission flow. This will ensure that any changes in the ZK Registry are reflected in the `RegistrationSMTReplicator` contract.
+
+You have two options:
+1. **Option A**: Set up a relayer service that listens for events from the ZK Registry and submits state updates to your `RegistrationSMTReplicator` instance.
+
+   - This relayer will listen for events such as new passport registrations or updates to existing registrations.
+   - When an event is detected, the relayer will call the `updateState(...)` function on the `RegistrationSMTReplicator` contract.
+   - This option is more user-friendly but requires you to pay the gas for the state updates on your chain.
+
+1. **Option B**: Send the updates the state update transactions on-demand from your front-end:
+   - This option is more cost-effective but requires the user to pay the gas for the state updates on your chain.
+   - You can use the `updateState(...)` function on the `RegistrationSMTReplicator` contract to submit state updates.
+
+
+To set up a relayer for the **Option A**, follow the instructions in <OutLink href="https://github.com/rarimo/proof-verification-relayer">proof-verification-relayer README</OutLink>.
```

Comment:
Option B

Reply:
Everything is ok here. Option A - relayer, Option B - updates from the front-end